brownie/_cli/analyze.py

```diff
@@ -37,179 +40,217 @@
+        self.reports: Dict[str, DetectedIssuesResponse] = {}
+        self.build = build
+        self.client = client or self.get_mythx_client()
+        self.highlight_report: Dict[str, dict] = {"highlights": {"MythX": {}}}
+        self.stdout_report: Dict[str, dict] = {}
 
+    @staticmethod
+    def get_mythx_client():
+        """Generate a MythX client instance."""
 
-def get_contract_types(build):
-    contracts = set([n for n, d in build.items() if d["type"] == "contract"])
-    libraries = set([n for n, d in build.items() if d["type"] == "library"])
+        if ARGV["api-key"]:
+            auth_args = {"api_key": ARGV["api-key"]}
+        elif environ.get("MYTHX_API_KEY"):
+            auth_args = {"api_key": environ.get("MYTHX_API_KEY")}
+        else:
+            raise ValidationError(
+                "You must provide a MythX API key via environment variable or the command line"
+            )
 
-    return contracts, libraries
+        return Client(
+            **auth_args, middlewares=[ClientToolNameMiddleware(name=f"brownie-{__version__}")]
+        )
 
+    def prepare_requests(self):
+        """Transform an artifact into a MythX payload."""
+
+        contracts = {n: d for n, d in self.build.items() if d["type"] == "contract"}
+        libraries = {n: d for n, d in self.build.items() if d["type"] == "library"}
+
+        requests = {}
+        for contract, artifact in contracts.items():
+            requests[contract] = self.construct_request_from_artifact(artifact)
+
+        # update requests with library dependencies
+        for library, artifact in libraries.items():
+            library_dependents = set(self.build.get_dependents(library))
+            contract_dependencies = set(contracts.keys()).intersection(library_dependents)
+            for contract in contract_dependencies:
+                requests[contract].sources.update(
+                    {
+                        artifact.get("sourcePath"): {
+                            "source": artifact.get("source"),
+                            "ast": artifact.get("ast"),
+                        }
+                    }
+                )
+
+        self.requests = requests
+
+    @classmethod
+    def construct_request_from_artifact(cls, artifact) -> AnalysisSubmissionRequest:
+        """Construct a raw submission request from an artifact JSON file."""
+
+        bytecode = artifact.get("bytecode", "")
+        deployed_bytecode = artifact.get("deployedBytecode", "")
+        source_map = artifact.get("sourceMap", "")
+        deployed_source_map = artifact.get("deployedSourceMap", "")
+
+        bytecode = re.sub(cls.BYTECODE_ADDRESS_PATCH, "0" * 40, bytecode)
+        deployed_bytecode = re.sub(cls.DEPLOYED_ADDRESS_PATCH, "0" * 40, deployed_bytecode)
+
+        source_list = artifact.get("allSourcePaths")
+        return AnalysisSubmissionRequest(
+            contract_name=artifact.get("contractName"),
+            bytecode=bytecode if bytecode else None,
```

Comment:
Lines 119-130, is there a reason to use a ternary operator rather than a boolean?  `bytecode or None` is a lot easier to read imo.

Reply:
No reason apart from taking over stale code :smile:  Fixed in 65dd5e0.